In [1]:
import pandas as pd
import sqlite3
from tqdm.notebook import tqdm

In [19]:
pd.options.display.max_columns = None
pd.options.display.max_rows = None

In [20]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
    SELECT *
    FROM nash_nppes
"""

nash_nppes = pd.read_sql(query,db)

In [21]:
db.close()

In [22]:
nash_nppes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38148 entries, 0 to 38147
Data columns (total 14 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   npi                  38148 non-null  int64  
 1   entity_type_code     38148 non-null  float64
 2   org_name             7466 non-null   object 
 3   last_name            30681 non-null  object 
 4   first_name           30682 non-null  object 
 5   name_suffix          895 non-null    object 
 6   provider_credential  25175 non-null  object 
 7   address_1            38148 non-null  object 
 8   zip                  38148 non-null  object 
 9   taxonomy_code        38148 non-null  object 
 10  code                 38148 non-null  object 
 11  grouping             38148 non-null  object 
 12  classification       38148 non-null  object 
 13  specialization       15120 non-null  object 
dtypes: float64(1), int64(1), object(12)
memory usage: 4.1+ MB


In [23]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
    SELECT COUNT(*)
    FROM hop_team
"""

hop_team = pd.read_sql(query,db)

db.close()

In [24]:
hop_team

COUNT(*)
0  31704890

### How many records in the hop_team table overlap with NPI numbers in nash_nppes?

In [34]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
   SELECT *
    FROM nash_nppes
    WHERE entity_type_code == 2  
"""

test = pd.read_sql(query,db)

db.close()

In [35]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7466 entries, 0 to 7465
Data columns (total 14 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   npi                  7466 non-null   int64  
 1   entity_type_code     7466 non-null   float64
 2   org_name             7466 non-null   object 
 3   last_name            0 non-null      object 
 4   first_name           0 non-null      object 
 5   name_suffix          0 non-null      object 
 6   provider_credential  0 non-null      object 
 7   address_1            7466 non-null   object 
 8   zip                  7466 non-null   object 
 9   taxonomy_code        7466 non-null   object 
 10  code                 7466 non-null   object 
 11  grouping             7466 non-null   object 
 12  classification       7466 non-null   object 
 13  specialization       2789 non-null   object 
dtypes: float64(1), int64(1), object(12)
memory usage: 816.7+ KB


In [36]:
#Checking Number of Records 
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
    SELECT COUNT(*)
    FROM hop_team
    WHERE from_npi IN(
        SELECT DISTINCT(npi)
        FROM nash_nppes
        WHERE entity_type_code == 1)    
    AND to_npi IN(
        SELECT DISTINCT(npi)
        FROM nash_nppes
        WHERE entity_type_code ==2)
"""

hops = pd.read_sql(query,db)

db.close()

In [ ]:
#query = """
#    SELECT *
#    FROM hop_team
#    WHERE from_npi IN(
#        SELECT DISTINCT(npi)
#        FROM nash_nppes)
#    AND to_npi IN(
#        SELECT DISTINCT(npi)
#        FROM nash_nppes)
#"""

In [37]:
hops

COUNT(*)
0     41127

There are 214,316 records in hop_team where the 'from_npi' NPI code matches one in the nash_nppes table

There are 214,080 records in hop_team where the 'to_npi' NPI code matches one in the nash_nppes table

In total, there are 138,493 hop_team records where both the 'to' and 'from' NPIs are in the nash_nppes table

If we filter down so that BOTH to/from npi are in Nashville and the from_npi to be entity type 1 and to_npi to be entity type 2, then there are 41,127 records.

### Add the filtered table to the database as 'nash_hops'

In [43]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
    SELECT *
    FROM hop_team
    WHERE from_npi IN(
        SELECT DISTINCT(npi)
        FROM nash_nppes
        WHERE entity_type_code == 1)
    AND to_npi IN(
        SELECT DISTINCT(npi)
        FROM nash_nppes
        WHERE entity_type_code ==2)
"""

nash_hops = pd.read_sql(query,db)

db.close()

In [44]:
nash_hops.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41127 entries, 0 to 41126
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   from_npi           41127 non-null  int64  
 1   to_npi             41127 non-null  int64  
 2   patient_count      41127 non-null  int64  
 3   transaction_count  41127 non-null  int64  
 4   average_day_wait   41127 non-null  float64
 5   std_day_wait       41127 non-null  float64
dtypes: float64(2), int64(4)
memory usage: 1.9 MB


In [45]:
#db = sqlite3.connect('../data/nppes_lite.sqlite') #open connection

#nash_hops.to_sql('nash_hops', db, if_exists = 'append', index = False) #save nash_hops data

#db.close() #close connection

In [2]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
    SELECT *
    FROM nash_hops
"""

test = pd.read_sql(query,db)

db.close()
test

from_npi      to_npi  patient_count  transaction_count  \
0       1023055126  1003013947             69                142   
1       1013154723  1003028770             32                 50   
2       1003963976  1003028770           2535               3945   
3       1023055126  1003031261             69                 99   
4       1013109511  1003075862            112                360   
...            ...         ...            ...                ...   
179615  1255575320  1245485549            914                958   
179616  1235137183  1245485549            198                253   
179617  1245541846  1245485549             95                111   
179618  1255428389  1245485549            191                236   
179619  1225389547  1245786482             66                 66   

        average_day_wait  std_day_wait  
0                  5.535        27.885  
1                 41.940        52.320  
2                  0.000         0.000  
3                 37.778        69.205  
4                 23.950        19.562  
...                  ...           ...  
179615             4.653        29.739  
179616            19.470        43.328  
179617            29.559        52.417  
179618             5.004        26.636  
179619            22.742        52.898  

[179620 rows x 6 columns]

In [3]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
    SELECT *
    FROM nash_nppes
"""

test = pd.read_sql(query,db)

db.close()
test

npi  entity_type_code  \
0      1134122187               1.0   
1      1043298722               1.0   
2      1255310058               1.0   
3      1740232081               1.0   
4      1962445585               1.0   
...           ...               ...   
38143  1659714087               1.0   
38144  1730671223               1.0   
38145  1376751099               2.0   
38146  1992223804               1.0   
38147  1164936878               1.0   

                                        org_name     last_name first_name  \
0                                           None      RUDNICKE     CHERYL   
1                                           None          CARO      SUSAN   
2                                           None         RUBLE      MAMIE   
3                                           None        BEADLE      TRACY   
4                                           None  GREEN-HADDEN        JOY   
...                                          ...           ...        ...   
38143                                       None      TASHJIAN      RANDY   
38144                                       None         LEWIS      ROGER   
38145  FRANK T RUTHERFORD MEMORIAL HOSPITAL, INC          None       None   
38146                                       None           RAY       TINA   
38147                                       None     BALENTINE    CHARLES   

      name_suffix provider_credential                  address_1    zip  \
0            None                CRNP             250 25TH AVE N  37203   
1            None      MSN, RNC, APNG             300 20TH AVE N  37203   
2            None                N.P.          3310 WEST END AVE  37203   
3            None                None              526 8TH AVE S  37203   
4            None                  NP          2400 PATTERSON ST  37203   
...           ...                 ...                        ...    ...   
38143        None                  MD           850 RS GASS BLVD  37216   
38144        None                None                717 HART LN  37216   
38145        None                None                 126 JMZ DR  38563   
38146        None                None    333 COMMERCE ST STE 700  37201   
38147        None                None  315 DEADERICK ST STE 1550  37238   

      taxonomy_code        code  \
0        363L00000X  363L00000X   
1        363L00000X  363L00000X   
2        363L00000X  363L00000X   
3        363L00000X  363L00000X   
4        363L00000X  363L00000X   
...             ...         ...   
38143    207ZF0201X  207ZF0201X   
38144    225C00000X  225C00000X   
38145    261QC0050X  261QC0050X   
38146    364SG0600X  364SG0600X   
38147    246QL0900X  246QL0900X   

                                                grouping  \
0      Physician Assistants & Advanced Practice Nursi...   
1      Physician Assistants & Advanced Practice Nursi...   
2      Physician Assistants & Advanced Practice Nursi...   
3      Physician Assistants & Advanced Practice Nursi...   
4      Physician Assistants & Advanced Practice Nursi...   
...                                                  ...   
38143                Allopathic & Osteopathic Physicians   
38144  Respiratory, Developmental, Rehabilitative and...   
38145                  Ambulatory Health Care Facilities   
38146  Physician Assistants & Advanced Practice Nursi...   
38147  Technologists, Technicians & Other Technical S...   

                           classification            specialization  
0                      Nurse Practitioner                      None  
1                      Nurse Practitioner                      None  
2                      Nurse Practitioner                      None  
3                      Nurse Practitioner                      None  
4                      Nurse Practitioner                      None  
...                                   ...                       ...  
38143                           Pathology        Forensic Pathology  
38144            

### Test join nash_nppes info to nash_hops for just from_npi

In [4]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
    SELECT *
    FROM nash_nppes
    LIMIT 10
"""

test = pd.read_sql(query,db)

db.close()

In [5]:
test

npi  entity_type_code org_name     last_name first_name name_suffix  \
0  1134122187               1.0     None      RUDNICKE     CHERYL        None   
1  1043298722               1.0     None          CARO      SUSAN        None   
2  1255310058               1.0     None         RUBLE      MAMIE        None   
3  1740232081               1.0     None        BEADLE      TRACY        None   
4  1962445585               1.0     None  GREEN-HADDEN        JOY        None   
5  1639199649               1.0     None      PETERSON  CHRISTINE        None   
6  1093736720               1.0     None       JACKSON      TAMMY        None   
7  1548274731               1.0     None    ANTANAITIS    MICHAEL        None   
8  1356434864               1.0     None        HURTIG       JANE        None   
9  1447344668               1.0     None        HRONEK        JAN        None   

  provider_credential          address_1    zip taxonomy_code        code  \
0                CRNP     250 25TH AVE N  37203    363L00000X  363L00000X   
1      MSN, RNC, APNG     300 20TH AVE N  37203    363L00000X  363L00000X   
2                N.P.  3310 WEST END AVE  37203    363L00000X  363L00000X   
3                None      526 8TH AVE S  37203    363L00000X  363L00000X   
4                  NP  2400 PATTERSON ST  37203    363L00000X  363L00000X   
5                  NP  832 WEDGEWOOD AVE  37203    363L00000X  363L00000X   
6                  NP     300 20TH AVE N  37203    363L00000X  363L00000X   
7                P.A.    1501 12TH AVE S  37203    363L00000X  363L00000X   
8                  NP     300 20TH AVE N  37203    363L00000X  363L00000X   
9                  NP     250 25TH AVE N  37203    363L00000X  363L00000X   

                                            grouping      classification  \
0  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
1  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
2  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
3  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
4  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
5  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
6  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
7  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
8  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   
9  Physician Assistants & Advanced Practice Nursi...  Nurse Practitioner   

  specialization  
0           None  
1           None  
2           None  
3           None  
4           None  
5           None  
6           None  
7           None  
8           None  
9           None

### Examine taxonomy

In [2]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
    SELECT *
    FROM nucc_taxonomy
"""

taxonomy = pd.read_sql(query,db)

db.close()

In [8]:
taxonomy.grouping.unique()

array(['Group', 'Allopathic & Osteopathic Physicians',
       'Behavioral Health & Social Service Providers',
       'Chiropractic Providers', 'Dental Providers',
       'Dietary & Nutritional Service Providers',
       'Emergency Medical Service Providers',
       'Eye and Vision Services Providers', 'Nursing Service Providers',
       'Nursing Service Related Providers', 'Other Service Providers',
       'Pharmacy Service Providers',
       'Physician Assistants & Advanced Practice Nursing Providers',
       'Podiatric Medicine & Surgery Service Providers',
       'Respiratory, Developmental, Rehabilitative and Restorative Service Providers',
       'Speech, Language and Hearing Service Providers',
       'Student, Health Care',
       'Technologists, Technicians & Other Technical Service Providers',
       'Agencies', 'Ambulatory Health Care Facilities', 'Hospital Units',
       'Hospitals', 'Laboratories', 'Managed Care Organizations',
       'Nursing & Custodial Care Facilities',


Exclude these? 

Under Grouping: 
        Laboratories, 
        Suppliers(? DME, Pharmacies, Prosthetic/Orthotic etc.), 
        Transportation Services,
        Group,
        
?Dental

Under Classification: Pharmacy

In [17]:
taxonomy[taxonomy.grouping=='Dental Providers']

code          grouping                      classification  \
281  125K00000X  Dental Providers  Advanced Practice Dental Therapist   
282  126800000X  Dental Providers                    Dental Assistant   
283  124Q00000X  Dental Providers                    Dental Hygienist   
284  126900000X  Dental Providers        Dental Laboratory Technician   
285  125J00000X  Dental Providers                    Dental Therapist   
286  122300000X  Dental Providers                             Dentist   
287  1223D0001X  Dental Providers                             Dentist   
288  1223D0004X  Dental Providers                             Dentist   
289  1223E0200X  Dental Providers                             Dentist   
290  1223G0001X  Dental Providers                             Dentist   
291  1223P0106X  Dental Providers                             Dentist   
292  1223X0008X  Dental Providers                             Dentist   
293  1223S0112X  Dental Providers                             Dentist   
294  1223X2210X  Dental Providers                             Dentist   
295  1223X0400X  Dental Providers                             Dentist   
296  1223P0221X  Dental Providers                             Dentist   
297  1223P0300X  Dental Providers                             Dentist   
298  1223P0700X  Dental Providers                             Dentist   
299  122400000X  Dental Providers                           Denturist   
300  125Q00000X  Dental Providers                     Oral Medicinist   

                               specialization  \
281                                      None   
282                                      None   
283                                      None   
284                                      None   
285                                      None   
286                                      None   
287                      Dental Public Health   
288                  Dentist Anesthesiologist   
289                               Endodontics   
290                          General Practice   
291          Oral and Maxillofacial Pathology   
292          Oral and Maxillofacial Radiology   
293            Oral and Maxillofacial Surgery   
294                            Orofacial Pain   
295  Orthodontics and Dentofacial Orthopedics   
296                       Pediatric Dentistry   
297                              Periodontics   
298                            Prosthodontics   
299                                      None   
300                                      None   

                                            definition effective_date  \
281  An Advanced Practice Dental Therapist is:<br><...      10/1/2012   
282  An individual who may or may not have complete...       4/1/2002   
283  An individual who has completed an accredited ...       4/1/2002   
284  An individual who has the skill and knowledge ...       4/1/2002   
285  A Dental Therapist is an individual who has co...      10/1/2012   
286  A dentist is a person qualified by a doctorate...       4/1/2002   
287  The science and art of preventing and controll...       4/1/2002   
288  A dentist who has successfully completed an ac...       4/1/2013   
289  The branch of dentistry that is concerned with...       4/1/2002   
290  A general dentist is the primary dental care p...       4/1/2002   
291  The specialty of dentistry and discipline of p...       4/1/2002   
292  The specialty of dentistry and discipline of r...       4/1/2002   
293  The specialty of dentistry which includes the ...       4/1/2002   
294  A dentist who assesses, diagnoses, and treats ...      10/1/2019   
295  That area of dentistry concerned with the supe...       4/1/2002   
296  An age-defined specialty that provides both pr...       4/1/2002   
297  That specialty of dentistry which encompasses ...       4/1/2002   
298  That branch of dentistry pertaining to the res...       4/1/2002   
299                                Definition to come.      

In [ ]:
db = sqlite3.connect('../data/nppes_lite.sqlite') #reopen the connection

query = """
SELECT 
    from_npi,
    
FROM nash_hops
"""

taxonomy = pd.read_sql(query,db)

db.close()